sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -754,6 +754,8 @@ case class AlterTableSetLocationCommand(
         // No partition spec is specified, so we set the location for the table itself
         catalog.alterTable(table.withNewStorage(locationUri = Some(location)))
     }
+
+    catalog.refreshTable(table.identifier)
```

Comment:
sorry, the test case hit the bug, so I fix it here, I will avoid the bug to use clear cache.